app/routes.js

```diff
@@ -86,12 +86,29 @@ function add_route(app_name) {
 
     log.debug(route.name + ': ' + route.pattern);
 
-    exports.router.all(route.pattern, route.view);
+    if (!route.method) {
+      route.method = 'get';
+    }
+
+    route.method = route.method.toLowerCase();
+
+    if (valid_method(route.method)) {
+      exports.router[route.method](route.pattern, route.view);
+
+    } else {
+      throw new Error('Invalid method "' + route.method.toUpperCase() + '" in route ' + route);
+    }
 
     return route;
   };
 }
 
+
+function valid_method(method) {
```

Comment:
In Ruby there is this amazing thing that you can end methods with question marks, e.g. `valid_method?` and you then know it would return a "boolean"...

...in absense of that I would probably call this `is_valid_method()` to make explicit it's returning a boolean.
I know it doesn't seem like a huge difference, especially when you looking at the function implementation, but it can helps avoiding confusion.

Reply:
Yeah, I'll change it. I personally don't like the flow of `if (is_valid_method(method)) {`, it sounds better as `if (valid_method(method))` or `if (is_valid(method))` - but the latter removes too much information from the function name.
Mind you, there's no guarantee that a function named `is_valid_method` will return a boolean, so naming conventions like that just feel like a step towards Hungarian notation (eg: `boolIsValidMethod`).

Reply:
Just reminded myself that I'm naming things with snake_case when they should really be camelCase in Javascript 😕

Reply:
Should they :trollface: ?

I know what you mean, I more used and prefer snake_case, and also when using different style in backend/frontend involves more work to convert cases it just becomes madness.